### PR TITLE
Add Client Well-Known docs.

### PR DESCRIPTION
**Pull request by pixlwave**
Following on from https://github.com/vector-im/element-ios/pull/7685, this PR adds documentation for all of the [`VectorWellKnown`](https://github.com/vector-im/element-ios/blob/develop/Riot/Model/WellKnown/VectorWellKnown.swift) types handled in EI.

### CODE REVIEW

File: docs/client_well_known.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+```
+
+```
+enum KeyPreSharingStrategy: Int {
+    case none = 0,
+    case whenEnteringRoom = 1,
+    case whenTyping = 2
+}
+```
```
Comment (pixlwave) [inline docs/client_well_known.md:47]: Not sure if there's a better way to document these enums, seems fine to me (but then I write Swift all day 🙃)

Comment (Johennes) [reply docs/client_well_known.md:47]: Probably clear enough for any developer to grasp 👍

File: docs/client_well_known.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## Jitsi Configuration
+
+Jitsi configuration is keyed by `io.element.jitsi`. The following options are supported:
+
+| Key | Type | 🤖 | 🍎 | 🕸️ | Description |
+| - | - | - | - | - | - |
+| `preferredDomain` | `String` | ❓ | ✅ | ❓ | Use the specified server for Jitsi calls. |
+| `useFor1To1Calls` | `Bool` | ❓ | ✅ | ❓ | Use Jitsi for 1:1 calls (by default we use native Matrix calls). |
```
Comment (Johennes) [inline docs/client_well_known.md]: Web doesn't support any of these currently from what I can see. Maybe you could fill that in while we're at it?

File: docs/client_well_known.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Encryption Configuration
+
+Encryption configuration is keyed by `io.element.e2ee`. The following options are supported:
+
+| Key | Type | 🤖 | 🍎 | 🕸️ | Description |
+| - | - | - | - | - | - |
+| `default` | `Bool` | ❓ | ✅ | ❓ | Indicate if E2EE is enabled by default. |
+| `secure_backup_required` | `Bool` | ❓ | ✅ | ❓ | Indicate if secure backup (SSSS) is mandatory. |
+| `secure_backup_setup_methods` | `BackupSetupMethod` | ❓ | ✅ | ❓ | Methods to use to setup secure backup (SSSS). |
```
Comment (Johennes) [inline docs/client_well_known.md]: Web supports this and the two above but _not_ the one below

Comment (Johennes) [reply docs/client_well_known.md]: However, web supports this in the same `io.element.e2ee` object:

```
    /**
     * Forces the encryption to disabled for all new rooms
     * When true, overrides configured 'default' behaviour
     * Hides the option to enable encryption on room creation
     * Disables the option to enable encryption in room settings for all new and existing rooms
     */
    force_disable?: boolean;
```

**Review by Johennes — APPROVED** [review]
If you have 2 minutes to add the web values from my comments, that'd be great. But otherwise looks amazing IMO. Thanks for doing this!

**Comment by pixlwave** [comment]
@Johennes nice, updated with the feedback. Let's see if we can get feedback for Android too before merging.

File: docs/client_well_known.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+
+## Jitsi Configuration
+
+Jitsi configuration is keyed by `io.element.jitsi`. The following options are supported:
+
+| Key | Type | 🤖 | 🍎 | 🕸️ | Description |
+| - | - | - | - | - | - |
+| `preferredDomain` | `String` | ❓ | ✅ | ❌ | Use the specified server for Jitsi calls. |
```
Comment (bmarty) [inline docs/client_well_known.md]: Supported on EA.

File: docs/client_well_known.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+| `useFor1To1Calls` | `Bool` | ❓ | ✅ | ❌ | Use Jitsi for 1:1 calls (by default we use native Matrix calls). |
```
Comment (bmarty) [inline docs/client_well_known.md]: Not supported on EA.

File: docs/client_well_known.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+## Encryption Configuration
+
+Encryption configuration is keyed by `io.element.e2ee`. The following options are supported:
+
+| Key | Type | 🤖 | 🍎 | 🕸️ | Description |
+| - | - | - | - | - | - |
+| `default` | `Bool` | ❓ | ✅ | ✅ | Indicate if E2EE is enabled by default. |
```
Comment (bmarty) [inline docs/client_well_known.md]: Supported by EA.

File: docs/client_well_known.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+| `force_disable` | `Bool` | ❓ | ❌ | ✅ | Overrides `default` when true, removing the option to enable encryption throughout the UI (existing encrypted rooms are unaffected). |
```
Comment (bmarty) [inline docs/client_well_known.md]: Not supported by EA.

File: docs/client_well_known.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+| `secure_backup_required` | `Bool` | ❓ | ✅ | ✅ | Indicate if secure backup (SSSS) is mandatory. |
```
Comment (bmarty) [inline docs/client_well_known.md]: Supported by EA.

File: docs/client_well_known.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+| `secure_backup_setup_methods` | `BackupSetupMethod` | ❓ | ✅ | ✅ | Methods to use to setup secure backup (SSSS). |
```
Comment (bmarty) [inline docs/client_well_known.md]: Supported by EA.

File: docs/client_well_known.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+| `outbound_keys_pre_sharing_mode` | `KeyPreSharingStrategy` | ❓ | ✅ | ❌ | Outbound keys pre sharing strategy. |
```
Comment (bmarty) [inline docs/client_well_known.md]: Supported by EA.

**Review by bmarty — COMMENTED** [review]
Thanks for adding this doc!

I have added comments about what is supported on EA.

EA also supports `"m.tile_server"."map_style_url"`, which is not documented yet, if I am not mistaken.

Code (model) on EA: https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/raw/wellknown/ElementWellKnown.kt#L23

**Comment by pixlwave** [comment]
Super, thanks @bmarty, updated with your feedback. For the tile server, I think the documentation within the MSC should probably suffice? https://github.com/matrix-org/matrix-spec-proposals/blob/matthew/location/proposals/3488-location.md#well-known-configuration

**Comment by bmarty** [comment]
> For the tile server, I think the documentation within the MSC should probably suffice? https://github.com/matrix-org/matrix-spec-proposals/blob/matthew/location/proposals/3488-location.md#well-known-configuration

I was not sure if the idea of this doc was to centralize all the possible configurations for the .well-known file, that Element clients may understand. If this is the case, I would add a paragraph about map tiler, which can eventually contain a link to the MSC.

[Element iOS](https://github.com/vector-im/element-ios/blob/develop/Riot/Model/WellKnown/VectorWellKnown.swift) does not seems to support the location configuration, that's why I though this was forgotten.

**Comment by pixlwave** [comment]
Ah, EI does support the location configuration but its all defined in the SDK, so in the app it is handled in the [configuration builder](https://github.com/vector-im/element-ios/blob/f1b6f52fde85ab665e10f48db474acc8047d1711/Riot/Model/HomeserverConfiguration/HomeserverConfigurationBuilder.swift#L79-L87) alongside the use of the VectorWellKnown models.

My original intent of this doc was to a place to capture Element specific configuration options that aren't documented anywhere outside of the code. But I can see where you're coming from, as EI also supports the [device dehydration MSC](https://github.com/vector-im/element-ios/blob/f1b6f52fde85ab665e10f48db474acc8047d1711/Riot/Model/HomeserverConfiguration/HomeserverConfigurationBuilder.swift#L54) in the well-known too. Maybe the best thing here would be to link to the respective parsers for each platform so its easy to find the implementations for the spec/msc documented support too?

We can do that in a follow-up though, for now I'm going to merge this PR.

**Pull request merged**